### DISCUSSION TITLE
How to set the Total Energy to align with the S0 source

### DISCUSSION THREAD

**Discussion by andy778** [klaasnicolaas/home-assistant-glow#190] [General]
Would it be possible to sync the value from the meter from the esphome web interface? I was thinking to add a section similar to rest button, but how to fetch a value... or am I totally wrong in my thinking? 

![image](https://user-images.githubusercontent.com/10696501/184931985-3e37128c-b34a-47eb-9e1e-d8c396f7d1f2.png)
![powermeter](https://user-images.githubusercontent.com/10696501/184932215-afe35507-03b2-4215-a0a1-6936f02e5dfe.jpg)

**Comment by andy778** [comment]
Basic use case is to see how much the recognized S0 pulses drifts  compared to the real meter

**Comment by klaasnicolaas** [comment]
I don't think there is an option yet to enter a value in the web-interface of an esphome node.

**Reply by andy778** [reply]
OK, then I was thinking wrong ( I can figure out some other way)  e.g.
- Compile in the start value of esphome
- Set the value perhaps from homeassitant 
- ...

but any way answer to my question, thanks case closed

**Comment by PatricDahl** [comment]
Hi!

Have similar query, is it possible to change Total Energy value from inside of Home Assistant?

What I have found out with my basic understanding is that if i change value to in my example 5000 and then call below I get 5Kwh

service: esphome.sensor_energy_monitoring_house_reset_total_energy
data: {}

button:
  - platform: restart
    name: "Restart - Glow"
  - platform: template
    name: "Reset - Total Energy"
    id: button_reset_total
    on_press:
      then:
      - pulse_meter.set_total_pulses:
          id: energy_pulse_meter
          value: 5000  

My goal is same to get meter on wall inline with value in Home assistant.

Thanks in advance
Patric

**Reply by klaasnicolaas** [reply]
It is unknown to me how you can easily run this over a service, personally my approach would be to see if it goes through the web UI of the glow. But so far I know ESPHome still doesn't support a number/text input field 🤔
